src/layer/lstm.cpp

```diff
@@ -240,4 +230,32 @@ int LSTM::forward(const Mat& bottom_blob, Mat& top_blob, const Option& opt) cons
     return 0;
 }
 
+int LSTM::forward(const std::vector<Mat>& bottom_blobs, std::vector<Mat>& top_blobs, const Option& opt) const
+{
+    const Mat& bottom_blob = bottom_blobs[0];
+
+    int T = bottom_blob.h;
+    Mat& top_blob = top_blobs[0];
+    Mat& hidden_state = top_blobs[1];
+    Mat& cell_state = top_blobs[2];
```

Comment:
hidden and cell are optional inputs/outputs
check bottom_blobs.size() and top_blobs.size() and ```return forward(bottom_blob, top_blob, opt);``` when necessary